src/main/java/bisq/core/payment/payload/MoneyGramAccountPayload.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.payment.payload;
+
+import bisq.core.locale.BankUtil;
+import bisq.core.locale.CountryUtil;
+
+import io.bisq.generated.protobuffer.PB;
+
+import com.google.protobuf.Message;
+
+import org.springframework.util.CollectionUtils;
+
+import java.nio.charset.Charset;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+
+@EqualsAndHashCode(callSuper = true)
+@ToString
+@Setter
+@Getter
+@Slf4j
+public class MoneyGramAccountPayload extends CountryBasedPaymentAccountPayload {
+    private String holderName;
+    private String city;
+    private String state = ""; // is optional. we don't use @Nullable because it would makes UI code more complex.
+    private String email;
+
+    public MoneyGramAccountPayload(String paymentMethod, String id) {
+        super(paymentMethod, id);
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // PROTO BUFFER
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private MoneyGramAccountPayload(String paymentMethodName,
+                                    String id,
+                                    String countryCode,
+                                    String holderName,
+                                    String city,
+                                    String state,
+                                    String email,
+                                    long maxTradePeriod,
+                                    @Nullable Map<String, String> excludeFromJsonDataMap) {
+        super(paymentMethodName,
+                id,
+                countryCode,
+                maxTradePeriod,
+                excludeFromJsonDataMap);
+        this.holderName = holderName;
+        this.city = city;
+        this.state = state;
+        this.email = email;
+    }
+
+    @Override
+    public Message toProtoMessage() {
+        PB.MoneyGramAccountPayload.Builder builder =
+                PB.MoneyGramAccountPayload.newBuilder()
+                        .setHolderName(holderName)
+                        .setCity(city)
+                        .setState(state)
```

Comment:
State is optional and need to be treated in PB optional otherwise it can cause errors. Is the same in WesternUnion (my fault).